tests/pos/depfuntype.scala

```diff
@@ -15,4 +15,21 @@ object Test {
   val d: C = c
   val z = depfun3(d)
   val z1: d.M = z
+
+  // Reproduced here because the one from DottyPredef is lacking a Tasty tree and
+  // therefore can't be inlined when testing non-bootstrapped.
+  // But inlining `implicitly` is vital to make the definition of `ifun` below work.
+  inline final def implicitly[T](implicit ev: T): T = ev
```

Comment:
The fact that adding `inline` changes how a method is typechecked doesn't seem like a great thing. I think @OlivierBlanvillain has some examples where it breaks working code.

Reply:
I am a bit nervous about changing it to `ev.type`. The inferencer has to do major acrobatics to deal with implicit method types. It might interfere with something else. The curent type of `implicitly` is simpler and clearer.